pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerServiceException.java

```diff
@@ -77,12 +77,6 @@ public TopicClosedException(Throwable t) {
         }
     }
 
-    public static class AddEntryMetadataException extends BrokerServiceException {
```

Comment:
I see that this isn't removed, we should deprecate the class.

Reply:
There is no other code that uses this exception.

Reply:
This might not be thrown by Pulsar, but a Pulsar Protocol handler could try to catch it. Therefore, I think we need to mark it as deprecated before we remove it.

Reply:
@michaeljmarshall For the protocol handler, it should not touch this exception. Here is an example https://github.com/apache/pulsar/pull/9039/files#diff-e7b41a1d0ffec3009ff825684ca865095884454d29d81b8f0edf6a6c129e21f4R26-R53 for adding a new entry metadata interceptor, I don't think a protocol handler will touch this exception unless the protocol handler implements it in the wrong way.  This is the part of the implementation that was forgotten to delete at the time (in 2.8.0). I have no objection with deprecating it first, just feel like we're adding complexity to something that was should be removed and never used.